base/src/numbers.rs

```diff
@@ -175,33 +154,38 @@ pub struct BlockTimeSpan(pub u64);
 #[derive(Clone, Serialize, Deserialize, Default, PartialEq, Debug)]
 pub struct BigInt256(pub [u8; 32]);
 
-impl From<BigInt256> for ark_ff::BigInteger256 {
-    fn from(i: BigInt256) -> Self {
-        use ark_ff::bytes::FromBytes;
-        Self::read(&i.0[..]).unwrap()
+impl AsRef<[u8]> for BigInt256 {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
     }
 }
 
-#[cfg(test)]
-pub mod tests {
-    use crate::numbers::Amount;
-    use crate::numbers::BigInt256;
-
-    #[test]
-    pub fn test_amount_to_formatted_string() {
-        assert_eq!(Amount(0).to_formatted_string(), "0.000000000");
-        assert_eq!(Amount(3).to_formatted_string(), "0.000000003");
-        assert_eq!(Amount(1000000003).to_formatted_string(), "1.000000003");
-        assert_eq!(Amount(1000000030).to_formatted_string(), "1.000000030");
-        assert_eq!(Amount(1300000000).to_formatted_string(), "1.300000000");
-        assert_eq!(Amount(1000000000).to_formatted_string(), "1.000000000");
+impl HexEncodable for BigInt256 {
+    type Error = hex::FromHexError;
+
+    fn to_hex_string(&self) -> String
+    where
+        Self: AsRef<[u8]>,
+    {
+        hex::encode(self)
+    }
+
+    fn try_from_hex(s: impl AsRef<[u8]>) -> Result<Self, Self::Error> {
+        let mut s = s.as_ref();
+        if s[1] == b'x' && (s[0] == b'0' || s[0] == b'\\') {
```

Comment:
I think a refactor would be possible here. I doesn't seem right to be to have the parsing of strings (which would be the same for all hex strings) in the `impl` specific to `BigInt256`

Reply:
Sry but did not quite get ur point, this is implementing HexEncodable trait for BigInt256 not for the struct itself, or do you suggest not implementing the trait? Could you plz elaborate on what would be a better design here?

Reply:
In both impls of `try_from_hex` you copy the code

```rust
if s[1] == b'x' && (s[0] == b'0' || s[0] == b'\\') {
    s = &s[2..];
}
```
It would be nice not to have this repeated.

I think it would make more sense to assume that the input slice only contains the actual hex of the data structure (not string encoded with a `0x` prefix) and handle stripping the prefix in the calling code. Possible refactor the prefix stripping to a function

Reply:
> Possible refactor the prefix stripping to a function

Fixed

> handle stripping the prefix in the calling code.

There're tons of occurrence for the hardcoded values with either 0x prefix (in json) or \x prefix (in ocaml code),  I don't feel it's worth the effort to manually strip those prefixes during copy & pasting, and we will need to do all those again when we define genesis config for devnet, does that make sense?